frontend/catalyst/cuda/catalyst_to_cuda_interpreter.py

```diff
@@ -832,75 +818,38 @@ def cudaq_backend_info(device, _config) -> BackendInfo:
+                warnings.warn(msg, UserWarning)
 
-    Returns:
-       wrapped: A wrapped function that will do the tracing.
-    """
+            self.jaxpr, self.out_treedef, self.c_sig = self.capture(args)
 
-    # TODO(@erick-xanadu): kwargs?
-
-    @wraps(fun)
-    def wrapped(*args, **_kwargs):
-        if _kwargs:
-            # TODO(@erick-xanadu):
-            raise NotImplementedError("CUDA tapes do not yet have kwargs.")  # pragma: no cover
-        # QJIT_CUDAQ(fun).get_jaxpr
-        # will return the JAXPR of the function fun.
-        # However, notice that *args are still concrete.
-        # This means that when we interpret the JAXPR, it will be a concrete interpretation.
-        # If we wanted an abstract interpretation, we could pass the abstract values that
-        # correspond to the return value of trace_to_jaxpr out_type2.
-        # If we did that, we could cache the result JAXPR from this function
-        # and evaluate it each time the function is called.
-        catalyst_jaxpr_with_host, out_tree = QJIT_CUDAQ(fun).get_jaxpr(*args)
-
-        # We need to keep track of the consts...
-        consts = catalyst_jaxpr_with_host.consts
-        catalyst_jaxpr = remove_host_context(catalyst_jaxpr_with_host)
-        # TODO(@erick-xanadu): There was a discussion about removing as much as possible the
-        # reliance on unstable APIs.
-        # Here is jax._src.core.ClosedJaxpr which is another function in the unstable API.
-        # Its only use here is to create a new ClosedJaxpr from the variable catalyst_jaxpr typed
-        # Jaxpr.
-        # Please note that catalyst_jaxpr_with_host is a ClosedJaxpr.
-        # But to get it without the host context we need to "open" it up again.
-        # And then close it again.
-        # pylint: disable-next=line-too-long
-        # From the [documentation (paraphrased)](https://jax.readthedocs.io/en/latest/jaxpr.html#understanding-jaxprs)
-        #
-        #    a closed jaxpr is has two fields the jaxpr and a list of constants.
-        #
-        # So, a good solution to get rid of this call here is to just interpret the host context.
-        # This is not too difficult to do. The only changes would be that we now need to provide
-        # semantics for func_p.
-        closed_jaxpr = jax._src.core.ClosedJaxpr(catalyst_jaxpr, catalyst_jaxpr.constvars)
+        # Not entirely sure how to determine if 'type promotion' is needed, or if the default
+        # type promotion value should be True or False.
+        return False  # no type promotion
 
-        # Because they become args...
-        args = list(args) + consts
-        args = tuple(args)
-        ctx = InterpreterContext(closed_jaxpr.jaxpr, closed_jaxpr.literals, *args)
-        out = interpret_impl(ctx, closed_jaxpr.jaxpr)
+    def run(self, args, kwargs):
+        # We need to override run, because it currently tries to call self.compiled_function,
+        # which does not exist here.
```

Comment:
Customizing run works nicely here and is also short 👍 

If we wanted an alternative with better potential integration into the cache system (related to comment above), we could also just store the desired callable into the `self.compiled_function` attribute during the `compile` stage, something that can be invoked as `self.compiled_function(*args, **kwargs)`.

```py
def compile(self):
  def compiled_fn(*args, **kwargs):
    ctx = InterpreterContext(self.jaxpr.jaxpr, self.jaxpr.literals, *args)
    results = interpret_impl(ctx, self.jaxpr.jaxpr)
  return compiled_fn, None
```
(as an example)